core/arch/arm/plat-stm32mp1/conf.mk

```diff
@@ -249,6 +249,14 @@ $(call force,CFG_STM32_I2C,y)
 $(call force,CFG_STM32_GPIO,y)
 endif
 
+#SAES and CRYP cannot be register at the same time in the crypto framework
+#If CRYP and SAES are enable on STMP32MP13, disable CRYP for safety purpose
```

Comment:
I don't think it's a good idea to compile both drivers and let the DT choose.
Today the device tree in built in OP-TEE, so you know which peripheral will be enabled or not.
You are building OP-TEE for a dedicated device tree (not a list of device tree)

The user may try to enables CRYP and SAES and this will result in a crash of the plateform. The sooner the error the better I think.

SAES is DPA resistant (CRYP is not) that's why we enable it in OP-TEE for MP13.
Moreover, SAES is mandatory to use access the HUK of the SoC. Using CRYP instead of SAES would mean no HUK services in OP-TEE which reduce the feature set of OP-TEE.